LearningMissionSimulation/PlaygroundGavril.cs

```diff
@@ -237,7 +241,31 @@ public void CreateClassrooms(int classroomCount)
 
         public void AssignInstructorsToClassrooms()
         {
-            throw new NotImplementedException();
+
+            string action = "Assign Instructors To Classrooms";
+            ReportHeader(actionName: action);
+
+            if (classroomList.Count == 0)
+            {
+                ReportError(missingResource: "Instructor", failedAction: action);
+            }
+            else
+            {
+                foreach (var classroom in classroomList)
+                {
+                    if (ModuleInstructorDictionary.ContainsKey(classroom.Module.Id))
+                    {
+                        List<Instructor> instructorList;
+                        ModuleInstructorDictionary.TryGetValue(classroom.Module.Id, out instructorList);
```

Comment:
You have to check the return value of
ModuleInstructorDictionary.TryGetValue(classroom.Module.Id, out instructorList)
-- before using  instructorList